src/actions/bns-client.ts

```diff
@@ -0,0 +1,28 @@
+"use server";
+
+import { getOwner, fetchUserOwnedNames } from "bns-v2-sdk";
+import { env } from "@/env";
+
+export type NetworkType = "mainnet" | "testnet";
+
+export async function bnsNameToAddress(name: string): Promise<string | null> {
+  const network: NetworkType = getNetwork();
+  return getOwner({
+    fullyQualifiedName: name,
+    network,
+  });
```

Comment:
This seems to be using `api.bnsv2.com` (before falling back to the read only contract call). I'm wondering if instead we should replace this with a readonly contract call, to avoid the trust assumption.